parsely/src/test/java/com/parsely/parselyandroid/EngagementManagerTest.kt

```diff
@@ -0,0 +1,158 @@
+package com.parsely.parselyandroid
+
+import androidx.test.core.app.ApplicationProvider
+import java.util.Calendar
+import java.util.TimeZone
+import java.util.Timer
+import org.assertj.core.api.AbstractLongAssert
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.Assertions.within
+import org.assertj.core.api.Assertions.withinPercentage
+import org.assertj.core.api.MapAssert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+private typealias Event = MutableMap<String, Any>
+
+@RunWith(RobolectricTestRunner::class)
+@Suppress("UNCHECKED_CAST")
+internal class EngagementManagerTest {
+
+    private lateinit var sut: EngagementManager
+    private val tracker = FakeTracker()
+    private val parentTimer = Timer()
+    private val baseEvent: Event = mutableMapOf(
+        "action" to "heartbeat",
+        "data" to testData
+    )
+
+    @Before
+    fun setUp() {
+        sut = EngagementManager(
+            tracker,
+            parentTimer,
+            DEFAULT_INTERVAL_MILLIS,
+            baseEvent,
+            FakeIntervalCalculator()
+        )
+    }
+
+    @Test
+    fun `when starting manager, then record the correct event after interval millis`() {
+        // when
+        sut.start()
+        sleep(DEFAULT_INTERVAL_MILLIS)
+        val timestamp = now - THREAD_SLEEPING_THRESHOLD
+
+        // then
+        assertThat(tracker.events[0]).isCorrectEvent(
```

Comment:
Minor (🔍): Consider making this assertion more readable, just like you did with the test further below:

```kotlin
        assertThat(tracker.events[0]).isCorrectEvent(
            // Ideally: totalTime should be equal to DEFAULT_INTERVAL_MILLIS
            withTotalTime = { isCloseTo(DEFAULT_INTERVAL_MILLIS, withinPercentage(10)) },
            // Ideally: timestamp should be equal to System.currentTimeMillis() at the time of recording the event
            withTimestamp = { isCloseTo(timestamp, within(100L)) }
        )
```